### PR TITLE
Internal tracing for the prover

### PR DESCRIPTION
**Pull request by tizoc**
Required for https://github.com/MinaProtocol/mina/pull/12875

### CODE REVIEW

**Comment by tizoc** [comment]
Hi @fabrizio-m, if I rebase this over `master` will it be compatible with the mina code in the `berkeley` branch? (the current PR implements modifications on top of the commit pointed out by the submodule in the `berkeley` branch).

**Comment by tizoc** [comment]
Great, thank you.

**Comment by fabrizio-m** [comment]
there seems to be a conflict but is just one missing line.

File: internal-tracing/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+                #[derive(Debug, ocaml::IntoValue, ocaml::FromValue, ocaml_gen::Struct)]
+                pub struct CamlTraces(String);
+
+                impl From<Traces> for CamlTraces {
+                    fn from(t: Traces) -> Self {
+                        Self(t.to_string())
+                    }
+                }
+
+//                 #[derive(Debug, ocaml::IntoValue, ocaml::FromValue, ocaml_gen::Struct)]
```
[the remark this answers is not in the document]
Comment (tizoc) [reply internal-tracing/src/lib.rs:126]: None, but just in case cc @binier, were you going to use it for anything? otherwise I will delete it.

**Comment by tizoc** [comment]
Deleted the commented out code. Btw once you think it is ready, let me know if you want me to squash the commits to remove the intermediary changes, will just take a minute.

File: internal-tracing/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+                              Apache License
```
Comment (fabrizio-m) [inline internal-tracing/LICENSE:1]: why another one? can't we have just the one at the root?

**Comment by fabrizio-m** [comment]
I think it's ready, when it passes CI I will approve it, seems to be just clippy stopping it now.

**Comment by tizoc** [comment]
Sorry, missed this. Fixing now.

**Review by fabrizio-m — APPROVED** [review]
Feel free to merge.

**Pull request merged**